readme.md

```diff
@@ -112,6 +112,44 @@ users[1]     # third_user
 
 ```
 
+### Streaming Downloads
+This SDK makes it easy to fetch a response from a Looker API and hydrate it into a Ruby object.This convenience is great for working with configuration and administrative data. However, when the response is gigabytes of row data, pulling it all into memory doesn't work so well - you can't begin processing the data until after it has all downloaded, for example, and chewing up tons of memory will put a serious strain on the entire system - even crash it. 
+
+One solution to all this is to use streaming downloads to process the data in chunks as it is downloaded. Streaming requires a little more code to set up but the benefits can be significant. 
+
+To use streaming downloads with the Looker SDK, simply add a block to an SDK call. The block will be called with chunks of data as they are downloaded instead of the method returning a complete result.
+
+For example:
+
+```ruby 
+def run_look_to_file(look_id, filename, format, opts = {})
+  File.open(filename, 'w') do |file|
+    sdk.run_look(look_id, format, opts) do |data, progress|
+      file.write(data)
+      puts "Wrote #{data.length} bytes of #{progress.length} total"
+    end  
+  end  
+end  
+
+run_look_to_file(38, 'out.csv', 'csv', limit: 10000)
+```
+
+In the code above, `sdk.run_look` opens a statement block. The code in the block will be called with chunks of data. The output looks like this:
+```
+Wrote 16384 bytes of 16384 total
+Wrote 16384 bytes of 32768 total
+Wrote 7327 bytes of 40095 total
+Wrote 16384 bytes of 56479 total
+etc...
+```
+
+##### Streaming Caveats
```

Comment:
No. Still single threaded and synchronous. 
Thanks for adding docs!
For extra points you could mention the 'stop' method in the Progress object that allows the callback to terminate the request.... https://github.com/looker/looker-sdk-ruby/pull/50/files#diff-e325af41ace76a35289c506ea6d4f46aR292